packages/components/pie-modal/src/index.ts

```diff
@@ -211,6 +218,31 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
         `;
     }
 
+    /**
+     * Render leadingAction button depending on prop availability.
+     *
+     * 1. If the prop `leadingAction` is not provided, the button is not rendered.
+     * 2. If the prop `leadingAction` is provided but any of the optional properties
+     * are not provided, they fall back to their default values.
+     *
+     * @param action
+     * @private
+     */
+    private renderLeadingAction (action: { text: string; variant?: string; ariaLabel?: string }): TemplateResult {
+        const { text, variant = 'primary', ariaLabel } = action;
+
+        return html`
+              <pie-button
+                  variant="${variant}"
+                  aria-label="${ariaLabel}"
```

Comment:
have you tested that screen readers narrate the aria label correctly when applied to pie button?

Reply:
Hadn't checked but have now and seems to work fine for me using voiceover.